### PR TITLE
Tag category

### LINKED ISSUE #51: Tag Categories
So that we don't have everything under role all the time 😅 .
1. In `cofed/sql`, create a new table `categories` that has an `id` and `name`. Also add a `category_id` foreign key in the schema for `tags`.
2. Modify the GET endpoint for tags to return all categories and the tags under them. (an object mapping from category name -> list of tags under that category) probably makes the most sense.
3. Have FiltersPage and the modal on ProfilePage query this endpoint to get the correct category names/tags for the dropdowns!

### PR DESCRIPTION
**Pull request by ehao1**
Fixes #51 

## Summary of Changes

Ranon and I made the tag-categories dynamic. Now, new categories can easily be added and deleted.

We created an array of categories, where each category contains a name, the dropdown options, and values.

## Testing

We manually tested a couple of cases. Not totally through but should be ok.

### CODE REVIEW

File: db/cofed.sql [hunk 1]
```diff
@@ -51,12 +58,72 @@ GRANT ALL PRIVILEGES ON ALL SEQUENCES IN SCHEMA public TO root;
...
+        'test2'),
+    ('test3@gmail.com', 'password3', 'Test3', '123456789',
+        'Berkeley, CA3', 37.8715, -123.2730, 'test.com', 'test3', 'test3', 'test3', 'test3',
+        'test3'),
+    ('test4@gmail.com', 'password4', 'Test4', '123456789',
+        'Berkeley, CA2', 37.8715, -123.2730, 'test.com', 'test2', 'test2', 'test2', 'test2',
+        'test2'),
+    ('test5@gmail.com', 'password5', 'Test5', '123456789',
+        'Berkeley, CA3', 37.8715, -123.2730, 'test.com', 'test3', 'test3', 'test3', 'test3',
+        'test3');
```
Comment (rrrliu) [inline db/cofed.sql]: let's keep the SQL file free of coop insertions until our deliverable prep tomorrow!

File: db/cofed.sql [hunk 2]
```diff
@@ -51,12 +58,72 @@ GRANT ALL PRIVILEGES ON ALL SEQUENCES IN SCHEMA public TO root;
...
+    ('Producer' , 1),
+    ('Community', 1),
+    ('Delivery', 1),
+    ('Medicinal Fungi' , 2),
+    ('Animal Products', 2),
+    ('Herbs' , 2),
+    ('Spices' , 2),
+    ('1-10', 3),
+    ('10-50', 3),
+    ('50-100', 3);
```
Comment (rrrliu) [inline db/cofed.sql]: great!! let's also add a category `'Services'` with
- advocacy
- education
- food services
- purchasing and acquisition

and also maybe add a 100+ tag for the Coop Size category!

@leebianca lmk if there's any more missing

File: db/cofed.sql [hunk 3]
```diff
@@ -51,12 +58,72 @@ GRANT ALL PRIVILEGES ON ALL SEQUENCES IN SCHEMA public TO root;
...
+    (3, 3),
+    (3, 7),
+    (4, 2),
+    (4, 3),
+    (4, 7),
+    (4, 11),
+    (5, 2),
+    (5, 4),
+    (5, 8),
+    (5, 10);
```
Comment (rrrliu) [inline db/cofed.sql]: also remove these for now :)

**Review by rrrliu — COMMENTED** [review]
great stuff!! thanks so much for the refactor and getting all this logic to work seamlessly. y'all are truly full-stack developers now.

just have some quick fixes in the SQL file and make sure to fix your merge conflicts + any eslint errors that may come up!

**Review by rrrliu — DISMISSED** [review]
fix ur build issues, otherwise u should be good!

**Review by leebianca — APPROVED** [review]

**Pull request merged**